checker.go

```diff
@@ -6,6 +6,14 @@ import (
 	"strings"
 )
 
+type sign int
```

Comment:
Could you please explain the reason for this extra type? I know you are using it for calculateStore. However it seems a [FlagArgument](http://martinfowler.com/bliki/FlagArgument.html).